src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -430,6 +445,7 @@ def get_result_set(
         to_iter: bool = False,
         block: bool = True,
         data_type: _AsyncResultType = _AsyncResultType.ROW,
+        case_insensitive: bool = False,
```

Comment:
I wonder if we should name it `case_sensitive` and flip the default value? This way we don't need to reason through two negations.